server/utils/taskListUtils.ts

```diff
@@ -37,22 +38,25 @@ const getCompleteSectionCount = (sections: FormSections, applicationOrAssessment
   }).length
 }
 
-const taskLink = (task: Task, applicationOrAssessment: Application | Assessment, journeyType?: JourneyType): string => {
+const isAssessment = (applicationOrAssessment: Application | Assessment): boolean => {
```

Comment:
instead of `: boolean` I think you could do `: applicationOrAssessment is Assessment` as a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). 
I don't think there's a major advantage of doing it here as `id` is defined on both